### ISSUE TITLE
Expose plugin to python api

### ISSUE THREAD

**Issue by synthdatagit** [ydrive/EasySynth#43]
It would be very helpful if the plugin was also exposed through a python api.

That would help automate data generation via scripting, which could reduce errors etc.

**Comment by fpanjevic** [comment]
Is this what you have in mind https://docs.unrealengine.com/5.0/en-US/scripting-the-unreal-editor-using-python/?

**Comment by synthdatagit** [comment]
> Is this what you have in mind https://docs.unrealengine.com/5.0/en-US/scripting-the-unreal-editor-using-python/?

Yes I would like to use python to start easysynth, assign actors to semantic classes etc. 

I am able to change actor position, material etc. using python, but I am not sure how to control EasySynth via Python plugin

**Comment by zlou** [comment]
> > Is this what you have in mind https://docs.unrealengine.com/5.0/en-US/scripting-the-unreal-editor-using-python/?
> 
> Yes I would like to use python to start easysynth, assign actors to semantic classes etc.
> 
> I am able to change actor position, material etc. using python, but I am not sure how to control EasySynth via Python plugin

I am looking for something similar. I think another user case would be changing the lighting conditions and other settings in the scene etc as a data augmentation. Then if one could control EasySynth, this will make the generation process fully automative and make EasySynth very useful. It would be great if you could give some hint in this respect.

**Comment by NikolaJov96** [comment]
Unfortunately, we have not yet been able to successfully access plugin functionality through Python. I would expect it to be straightforward, and a matter of enabling some kind of an option, but still no luck.

Here is our post on the UE forum asking exactly that. Unfortunately, we received no responses.
https://forums.unrealengine.com/t/how-to-expose-plugin-methods-to-python-scripting-api/659381
